mlrun/package/packagers_manager.py

```diff
@@ -715,17 +717,24 @@ def _unpack_data_item(self, data_item: DataItem, type_hint: type):
     def _look_for_extra_data(
         key: str,
         artifacts: list[Artifact],
+        artifact_uris: dict,
         results: dict,
     ) -> Union[Artifact, str, int, float, None]:
         """
         Look for an extra data item (artifact or result) by given key. If not found, None is returned.
 
-        :param key:       Key to look for.
-        :param artifacts: Artifacts to look in.
-        :param results:   Results to look in.
+        :param key:             Key to look for.
+        :param artifacts:       Artifacts to look in.
+        :param artifact_uris:   Artifacts URIs to look in.
+        :param results:         Results to look in.
 
         :return: The artifact or result with the same key or None if not found.
         """
+        artifact_uris = artifact_uris or {}
+        for _key, uri in artifact_uris.items():
+            if key == _key:
+                return uri
```

Comment:
Why not retrieve the artifact itself with `get_store_resource`? now extra_data holds the uri instead of the artifact, and I didn’t see you handle this case in `link_packages`

Reply:
Good catch 👍